File: backend/src/main/kotlin/com/qnocks/gorelaxauthbackend/service/AuthService.kt
package com.qnocks.gorelaxauthbackend.service

import com.qnocks.gorelaxauthbackend.dto.RegisterRequest
import com.qnocks.gorelaxauthbackend.entity.User

interface AuthService {

    fun register(request: RegisterRequest)

    fun test(): List<User>
}

File: frontend/src/app/layouts/auth/services/auth.service.ts
import {Injectable} from '@angular/core';
import {BaseApiService} from '../../../core/services/base-api.service';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {ApiEndpoints} from '../../../core/constants/api-endpoints.constants';
import {RegisterRequest} from '../models/register-request';

@Injectable()
export class AuthService extends BaseApiService {

  constructor(
    protected httpClient: HttpClient) {
    super(httpClient);
  }

  register(request: RegisterRequest): Observable<void> {
    return super.post(ApiEndpoints.AUTH.REGISTER, request);
  }
}
